odoo/service/model.py

```diff
@@ -118,22 +119,30 @@ def _(src):
                     if key in inst.pgerror:
                         raise ValidationError(tr(registry._sql_error[key], 'sql_constraint') or inst.pgerror)
                 if inst.pgcode in (errorcodes.NOT_NULL_VIOLATION, errorcodes.FOREIGN_KEY_VIOLATION, errorcodes.RESTRICT_VIOLATION):
-                    msg = _('The operation cannot be completed, probably due to the following:\n- deletion: you may be trying to delete a record while other records still reference it\n- creation/update: a mandatory field is not correctly set')
+                    msg = _('The operation cannot be completed.\n\n')
                     _logger.debug("IntegrityError", exc_info=True)
                     try:
                         errortxt = inst.pgerror.replace('«','"').replace('»','"')
                         if '"public".' in errortxt:
-                            context = errortxt.split('"public".')[1]
-                            model_name = table = context.split('"')[1]
+                            context = re.findall(r'"public"\."(.*?)["\[]', errortxt)
+                            model_name, rmodel_name = context[0], context[-1]
```

Comment:
We don't need to scrape the error message anymore.
Odoo now requires `psycopg2 > 2.5`, which makes the "diagnostics API" available,
like in https://github.com/odoo/odoo/blob/12.0/odoo/models.py#L5608.
Please use this instead!